cmd/semaphore/daemon/client.go

```diff
@@ -12,14 +12,15 @@ import (
 	"github.com/jexia/semaphore/pkg/broker/logger"
 	"github.com/jexia/semaphore/pkg/broker/trace"
 	"github.com/jexia/semaphore/pkg/functions"
+	"github.com/jexia/semaphore/pkg/prettyerr"
 	"github.com/jexia/semaphore/pkg/transport"
 	"go.uber.org/zap"
 )
 
 // NewClient constructs a new Semaphore instance
 func NewClient(ctx *broker.Context, core semaphore.Options, provider providers.Options) (*Client, error) {
 	if ctx == nil {
-		return nil, errors.New("nil context")
+		return nil, prettyerr.PrettyError(errors.New("nil context"))
```

Comment:
would it be easier to use defer statement like this (not calling explicitly `prettyerr.PrettyError( ... )` every time)? :thinking: 
```go 
func NewClient(ctx *broker.Context, core semaphore.Options, provider providers.Options) (client *Client,err error) {
	defer func() {
		if err != nil {
			err = prettyerr.PrettyError(err)
		}
	}()

        .   .   .
}
```
or wrap it on the top level (where `NewClient` is called)?

Reply:
This could indeed be implemented the `run` function of all CLIs to simplify the implementation. (https://github.com/jexia/semaphore/blob/master/cmd/semaphore/daemon/daemon.go#L49-L79)

Reply:
This could indeed be implemented the `run` function of all CLIs to simplify the implementation. (https://github.com/jexia/semaphore/blob/master/cmd/semaphore/daemon/daemon.go#L49-L79)

Reply:
Thanks for the review @Alma-media  @jeroenrinzema .
I'm relatively new to Golang, thus did not know this approach. Thanks a lot for the suggestion. 
I have updated the code in PR